src/fluree/server/consensus/shared/create.clj

```diff
@@ -0,0 +1,42 @@
+(ns fluree.server.consensus.shared.create
+  "Shared functions for ledger creation across consensus mechanisms."
+  (:require [fluree.db.api :as fluree]
+            [fluree.db.util.log :as log]))
+
+(defn hash-from-id
+  "Extracts the hash portion from a commit ID.
+  Commit IDs have format: 'fluree:commit:sha256:b<base32-hash>'"
+  [commit-id]
+  (when (string? commit-id)
```

Comment:
It might be useful to break all of this out into a `commit-id?` function. That function would probably come in handy and be reused, and you could just do `(when (commit-id? commit-id) ...`